plugins/vendor/sso/services/SsoService.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * @service sso
+ * @package plugins.sso
+ * @subpackage api.services
+ */
+class SsoService extends KalturaBaseService
+{
+	/**
+	 * Adds a new sso configuration(vendorIntegration of type sso).
+	 *
+	 * @action add
+	 * @param KalturaSso $sso a new sso configuration
+	 * @return KalturaSso The new sso configuration
+	 * @throws KalturaSsoErrors::DUPLICATE_SSO
+	 * @throws KalturaSsoErrors::MISSING_MANDATORY_PARAMETER
+	 */
+	public function addAction(KalturaSso $sso)
+	{
+		$sso->validateParameters();
+		$sso->validateDuplication();
+
+		$newSso = new SsoVendorIntegration();

```

Comment:
toInsertableObject should return to you the db object no need to manually create it.